flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractFlinkResourceReconciler.java

```diff
@@ -267,7 +270,8 @@ protected abstract boolean reconcileSpecChange(
      * @return True if any further reconciliation action was taken.
      * @throws Exception Error during reconciliation.
      */
-    protected abstract boolean reconcileOtherChanges(FlinkResourceContext<CR> ctx) throws Exception;
+    protected abstract boolean reconcileOtherChanges(
+            FlinkResourceContext<CR> ctx, boolean requireScale) throws Exception;
```

Comment:
Let's pass the difftype as an argument instead of the confusing boolean param I think.